### PR TITLE
Fix time parser

### LINKED ISSUE #231: [PE-D][Tester A] Add Event Command - Giving an invalid date still adds event on a different date
When user uses add event command with an invalid date but the day being within the range 1 - 31, it rounds off to the nearest date and adds an event.

Step 1: Add a new event using the add event command with a date in a month that doesn't have those dates. E.g. 29 Feb 2022 and 30 Feb 2022

Result: The event is added but with the start date and end date being 28 Feb 2022

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: DINESH1201/ped#2

### LINKED ISSUE #237: [PE-D][Tester D] System identifies a clash in events even when event start time matches a different event's end time
As seen in the screenshot, the error that there is another event at the same time is shown even though the start time of the new event is same as the end time of the other event (technically no clash). Just a small flaw, could consider adjusting logic to check if start time is greater than the end time of any events (not greater than or equals to). If not user would have to input back-to-back events with timing for example from 12:00 to 13:00, then from 13:01 to 14:00, which is not very natural.

-------------
Labels: `severity.High` `type.FeatureFlaw`
original: yixianggg/ped#4

### LINKED ISSUE #273: [PE-D][Tester B] Event with overlapping datetime ranges
While the app itself prevents adding events tied to the same person with overlapping datetime ranges, editing the JSON file directly allows me to add events with overlapping datetime ranges.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: NgZiXin/ped#7

### LINKED ISSUE #274: [PE-D][Tester B] Start dateTime after End dateTime
While application prevents me from adding an event where start dateTime is after end dateTime, I can edit the JSON file directly and bypass this constraint.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: NgZiXin/ped#8

### PR DESCRIPTION
**Pull request by fusin02**
closes #231 , closes #237 , closes #273 , closes #274

### CODE REVIEW

**Review by wrjgold — COMMENTED** [review]
This might be violate the feature freeze put in place.
"Accepting supposedly invalid values (e.g., end date is earlier than start date; February 30th) is, while not ideal, not necessarily incorrect either (i.e., adopting a [garbage-in garbage-out](https://en.wikipedia.org/wiki/Garbage_in,_garbage_out) approach to input validation). However, if such data can make other things go haywire (e.g., crash the app, corrupt the data file), accepting them can be considered a bug, and fixed."

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 1]
```diff
@@ -81,6 +82,9 @@ public AddressBook toModelType() throws IllegalValueException {
             if (addressBook.hasEvent(event)) {
                 throw new IllegalValueException(MESSAGE_DUPLICATE_EVENT);
             }
+            if (addressBook.hasEventOverlap(event)) {
+                throw new IllegalValueException(MESSAGE_OVERLAPPING_EVENT);
```
Comment (seantham21) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:86]: This might not be allowed due to the feature freeze.

**Review by seantham21 — COMMENTED** [review]
LGTM, new UI Message can be discussed

**Review by wrjgold — APPROVED** [review]
nice thanks fusin

**Pull request merged**